odl/space/space_utils.py

```diff
@@ -22,35 +22,38 @@
 from future import standard_library
 standard_library.install_aliases()
 
-__all__ = ('vector', 'ntuples', 'fn', 'cn', 'rn')
-
 import numpy as np
 
 from odl.set import RealNumbers, ComplexNumbers
-from odl.util.utility import (
-    is_real_floating_dtype, is_complex_floating_dtype, is_scalar_dtype)
-from odl.space.entry_points import NTUPLES_IMPLS, FN_IMPLS
+from odl.util.utility import is_scalar_dtype
+from odl.space.entry_points import TENSOR_SET_IMPLS, TENSOR_SPACE_IMPLS
 
 
-def vector(array, dtype=None, impl='numpy'):
-    """Create an n-tuples type vector from an array.
+__all__ = ('tensor', 'tensor_set', 'tensor_space', 'ctensors', 'rtensors',
```

Comment:
I have a hard time seeing why we need all of these, i would simply overload the meaning of the old ones, i.e. remove `tensor`, `tensor_set`, `tensor_space`, `ctensors` and `rtensors`, and move their respective new functionality into the respective old method.

Anyone reading `odl.rn((3, 2))` would understand that it means, at least in my opinion. And we wouldn't clutter the top level namespace as much.